tests/test_tutorial.py

```diff
@@ -0,0 +1,396 @@
+      mock_logger.info.assert_called_with(
+          "Dirty roles: " + str(['snapshot', 'targets', 'timestamp']))
+
+    repository.writeall()
+
+    repository.targets.remove_target('myproject/file4.txt')
+    self.assertTrue(os.path.exists(os.path.join(
+        'repository','targets', 'myproject', 'file4.txt')))
+
+    with mock.patch("tuf.repository_tool.logger") as mock_logger:
+      repository.dirty_roles()
+      # Concat strings to avoid Python2/3 unicode prefix problems ('' vs. u'')
+      mock_logger.info.assert_called_with(
+          "Dirty roles: " + str(['targets']))
+
+    repository.mark_dirty(['snapshot', 'timestamp'])
+    repository.writeall()
+
+
+    # ----- Tutorial Section: Excursion: Dump Metadata and Append Signature
+    signable_content = dump_signable_metadata(
+        os.path.join('repository', 'metadata.staged', 'timestamp.json'))
+
+    # Skipping user entry of password
+    ## private_ed25519_key = import_ed25519_privatekey_from_file('ed25519_key')
+    private_ed25519_key = import_ed25519_privatekey_from_file('ed25519_key', 'password')
+    signature = create_signature(
+        private_ed25519_key, encode_canonical(signable_content).encode())
+    append_signature(
+        signature,
+        os.path.join('repository', 'metadata.staged', 'timestamp.json'))
+
+
+
+    # ----- Tutorial Section: Delegations
+    generate_and_write_rsa_keypair(
+        'unclaimed_key', bits=2048, password='password')
+    public_unclaimed_key = import_rsa_publickey_from_file('unclaimed_key.pub')
+    repository.targets.delegate(
+        'unclaimed', [public_unclaimed_key], ['myproject/*.txt'])
+
+    repository.targets("unclaimed").add_target("myproject/file4.txt")
+
+    # Skipping user entry of password
+    ## private_unclaimed_key = import_rsa_privatekey_from_file('unclaimed_key')
+    private_unclaimed_key = import_rsa_privatekey_from_file(
+        'unclaimed_key', 'password')
+    repository.targets("unclaimed").load_signing_key(private_unclaimed_key)
+
+
+    with mock.patch("tuf.repository_tool.logger") as mock_logger:
+      repository.dirty_roles()
+      # Concat strings to avoid Python2/3 unicode prefix problems ('' vs. u'')
+      mock_logger.info.assert_called_with(
+          "Dirty roles: " + str(['targets', 'unclaimed']))
+
+    repository.mark_dirty(["snapshot", "timestamp"])
+    repository.writeall()
+
+
+    # Simulate the following shell command:
+    ## $ cp -r "repository/metadata.staged/" "repository/metadata/"
+    shutil.copytree(
+        os.path.join('repository', 'metadata.staged'),
+        os.path.join('repository', 'metadata'))
+
+
+    # ----- Tutorial Section: Delegate to Hashed Bins
+
+    targets = repository.get_filepaths_in_directory(
+        os.path.join('repository', 'targets', 'myproject'), recursive_walk=True)
+
+    repository.targets('unclaimed').delegate_hashed_bins(
+        targets, [public_unclaimed_key], 32)
```

Comment:
Should we make a remark about the output here? I get:

```
Creating hashed bin delegations.
1 total targets.
32 hashed bins.
256 total hash prefixes.
Each bin ranges over 8 hash prefixes.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
Adding a verification key that has already been used.
```

Reply:
Yes! It is another instance of me thinking this should be addressed. More specifically with the 2nd to last item in https://github.com/theupdateframework/tuf/issues/808#issuecomment-487596129.

But adding a comment there that briefly explains the output does not hurt.